lib/stages/oneDiskVDIFWrite.cpp

```diff
@@ -0,0 +1,338 @@
+    uint32_t vdif_frame_size;
+    uint32_t nvdif_payload_per_file;
+    uint32_t in_buf_frame_size;
+    uint32_t src_start;
+    uint32_t target_start;
+    uint32_t inframe_ctr = 0;
+    long intsec_offset;
+    char setting_log_file[full_path_len];
+    char data_log_file[full_path_len];
+    FILE* setting_file;
+    FILE* j_file;
+
+    MergedBeamMetadata* in_metadata = nullptr;
+    timespec out_frame_unix_time;
+    timespec vdif_file_start_time;
+
+
+    time(&rawtime);
+
+    timeinfo = gmtime(&rawtime);
+    // The time prescision in the file should never be more than a second. 
+    // Since creating the directory is independent for each disk now.
+    // To make sure the directory name is the same, the precision of time stamp
+    // in the directory name should be limited to seconds.
+    strftime(data_time, sizeof(data_time), "%Y%m%dT%H%M%SZ", timeinfo);
+    snprintf(data_set_c, sizeof(data_set_c), "%s_%s_%s", data_time, instrument_name.c_str(),
+             file_ext.c_str());
+    
+    // Make output directory
+    if (write_to_disk) {
+        if (use_abs_path){
+            snprintf(data_dir_name, sizeof(data_dir_name), "%s", abs_path.c_str());
+	}
+	else{
+	    snprintf(data_dir_name, sizeof(data_dir_name), "%s/%s/%d/%s", disk_base.c_str(), disk_set.c_str(), disk_id, data_set_c);
+	}
+	make_dir(data_dir_name);
+	// write setting log
+	snprintf(setting_log_file, full_path_len, "%s/%s.txt", data_dir_name, "setting_log");
+        setting_file = fopen(setting_log_file, "w");
+	snprintf(data_log_file, full_path_len, "%s/%s.txt", data_dir_name, "data_log");
+	j_file = fopen(data_log_file, "w");
+    }
+    
+        
+    // allocate the vdif buffer 
+    vdif_frame_data_size = vdif_samples_per_frame * vdif_freq_per_frame * num_pol;
+    vdif_frame_size = vdif_frame_header_size + vdif_frame_data_size;
+    INFO("VDIF size {:d} {:d}", vdif_frame_data_size, vdif_frame_size);
+    vdif_frame = (uint8_t*) malloc(vdif_frame_size);
+    // Cast vdif header
+    vdif_header = (VDIFHeader*) &vdif_frame[0];
+    // cast vdif payload
+    payload = &vdif_frame[vdif_frame_header_size];
+    // Fill fixed parts of vdif frame header
+    vdif_header -> invalid = 0;
+    vdif_header -> legacy = 0;
+    // Only take the last two digits of the year.
+    // If the time beyond 2100, we should consider change the code here
+    vdif_header -> ref_epoch = (uint32_t)((ref_year - 2000) * 2) ; // Cast it to int. 
+    vdif_header -> unused = 0;
+    vdif_header -> log_num_chan = (int)log2 ((float)(vdif_freq_per_frame * num_pol));
+    vdif_header -> frame_len = (vdif_frame_data_size + vdif_frame_header_size) / 8;
+    vdif_header -> vdif_version = 0;
+    vdif_header -> data_type = 1;
+    vdif_header -> bits_depth = 4 - 1;  //  bits per sample minus 1
+    vdif_header -> station_id = 9999;
+    vdif_header -> edv = 0;   // for now
+    vdif_header -> eud1 = 0; 
+    vdif_header -> eud2 = 0;
+    vdif_header -> eud3 = 0;
+    vdif_header -> eud4 = 0;
```

Comment:
The formatting here doesn't follow our design guide, e.g. no spaces in member access `vdif_header->invalid`.  I think once you run clang-format, that should automatically fix this.